### PR TITLE
Moved StakedChunk cache invalidation into model object

### PR DESCRIPTION
**Pull request by angushtlam**
If we invalidate the cache on save before the find there wouldn't be an infinite loop

red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert 🚨 red diff alert

### CODE REVIEW

File: src/main/java/com/raeic/embarker/land/models/StakedChunk.java [hunk 1]
```diff
@@ -67,62 +55,41 @@ public void setLastUpdated(Timestamp lastUpdated) {
         this.lastUpdated = lastUpdated;
     }
 
-    public void save() {
+    public void delete() {
+        // Invalidate caches because it's no longer accurate on write
+        String stakedChunkCacheKey = coordX + "," + coordZ + "," + worldName;
+        Globals.stakedChunks.invalidateCacheByKey(stakedChunkCacheKey);
+        Globals.embarkerPlayers.invalidateCacheByKey(ownerUniqueId);
```
Comment (nenofite) [inline src/main/java/com/raeic/embarker/land/models/StakedChunk.java]: another one 🐎

File: src/main/java/com/raeic/embarker/land/models/StakedChunk.java [hunk 2]
```diff
@@ -67,62 +55,41 @@ public void setLastUpdated(Timestamp lastUpdated) {
...
         });
     }
 
-    public void delete() {
-        // Also, invalidate caches because it's no longer accurate on write
+    public void save() {
+        // Invalidate caches because it's no longer accurate on write
         String stakedChunkCacheKey = coordX + "," + coordZ + "," + worldName;
         Globals.stakedChunks.invalidateCacheByKey(stakedChunkCacheKey);
         Globals.embarkerPlayers.invalidateCacheByKey(ownerUniqueId);
```
Comment (nenofite) [inline src/main/java/com/raeic/embarker/land/models/StakedChunk.java]: ditto

File: src/main/java/com/raeic/embarker/land/models/StakedChunkManager.java [hunk 3]
```diff
@@ -80,6 +67,7 @@ public StakedChunk findOneIgnoreCache(int coordX, int coordZ, String worldName)
                     Timestamp firstStaked = results.getTimestamp("firstStaked");
                     Timestamp lastUpdated = results.getTimestamp("lastUpdated");
                     result = Globals.stakedChunks.create(coordX, coordZ, worldName, ownerUniqueId, firstStaked, lastUpdated);
+                    result.save();
```
Comment (nenofite) [inline src/main/java/com/raeic/embarker/land/models/StakedChunkManager.java:70]: I think writing to DB as the only way to insert into the cache is not great but if it works, it works \o/

**Review by nenofite — CHANGES_REQUESTED** [review]

File: src/main/java/com/raeic/embarker/land/models/StakedChunkManager.java (hunk 3 above)
Comment (angushtlam) [reply src/main/java/com/raeic/embarker/land/models/StakedChunkManager.java:70]: (:

**Review by nenofite — APPROVED** [review]

**Pull request merged**